templates/app/offline-worker.js

```diff
@@ -0,0 +1,129 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+<% importScripts.forEach(function (filepath) {
+%>importScripts('<%- filepath %>');
+<% }); %>
+(function (self) {
+  'use strict';
+
+  // On install, cache resources and skip waiting so the worker won't
+  // wait for clients to be closed before becoming active.
+  self.addEventListener('install', function (event) {
+    event.waitUntil(oghliner.cacheResources().then(function () {
+      if (typeof self.skipWaiting === 'function') {
+        return self.skipWaiting();
+      }
+    }));
+  });
+
+  // On activation, delete old caches and start controlling the clients
+  // without waiting for them to reload.
+  self.addEventListener('activate', function (event) {
+    event.waitUntil(oghliner.clearOtherCaches().then(function () {
+      if (self.clients && typeof self.clients.claim === "function") {
+        return self.clients.claim();
+      }
+    }));
+  });
+
+  // Retrieves the request following oghliner strategy.
+  self.addEventListener('fetch', function (event) {
+    if (event.request.method === 'GET') {
+      event.respondWith(oghliner.get(event.request));
+    }
+    else {
```

Comment:
Nit: cuddle the *else* keyword, i.e.:

    } else {

(I personally prefer elses uncuddled, the way you've written this one; but else-cuddling is conventional in this codebase, as it is in a bunch of other Mozilla codebases, and I'd rather we keep it consistent.)